models/DataObject/ClassDefinition/Data/IdRewriterInterface.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Model\DataObject\ClassDefinition\Data;
+
+interface IdRewriterInterface
+{
+    /**
+     * Rewrites id from source to target, $idMapping contains
+     * array(
+     *  "document" => array(
+     *      SOURCE_ID => TARGET_ID,
+     *      SOURCE_ID => TARGET_ID
+     *  ),
+     *  "object" => array(...),
+     *  "asset" => array(...)
+     * )
+     *
+     * @param mixed $object
+     * @param array $idMapping
+     * @param array $params
+     *
+     * @return array
+     */
+    public function rewriteIds($object, $idMapping, $params = []);
```

Comment:
I think the contract should be typed:

```suggestion
    public function rewriteIds(mixed $object, array $idMapping, array $params = []): array;
```